io-engine/src/core/reactor.rs

```diff
@@ -624,3 +635,101 @@ impl Future for &'static Reactor {
         }
     }
 }
+
+/// Maximum number of heartbeats a reactor is allowed to miss
+/// before it is classified as frozen.
+#[cfg(feature = "diagnostics")]
+const MAX_MISSED_HEARTBEATS: u64 = 3;
+
+/// Monitor health for all reactors: all available reactors are constantly
+/// monitored for liveness.
+#[cfg(feature = "diagnostics")]
+pub async fn reactor_monitor_loop() {
+    use std::sync::atomic::{AtomicU64, Ordering};
+
+    /// Metadata for every reactor being monitored by the reactor monitor.
+    struct ReactorRecord {
+        frozen: bool,
+        reactor: &'static Reactor,
+        reactor_tick: &'static AtomicU64,
+        core: u32,
+    }
+
+    let mut interval = tokio::time::interval(Duration::from_secs(1));
+    let mut tick: u64 = 0;
+    let mut reactor_state: Vec<ReactorRecord> =
+        Vec::with_capacity(Cores::count().id() as usize);
```

Comment:
shouldn't this be `Cores::count() as usize` ?

Reply:
::count() has a bit different semantics in our SPDK implementation:

```    pub fn count() -> Self {
        Cores(Self::get_core(Core::Count))    
}
```

The closest possible solution for us is:
```
Cores::get_core(Core::Count) as usize
```

Reply:
Thanks for explaining :+1: